pkg/infrastructure/powervs/clusterapi/powervs.go

```diff
@@ -178,6 +178,7 @@ func (p Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput)
 			Steps:    math.MaxInt32}
 		err = wait.ExponentialBackoffWithContext(ctx, backoff, func(context.Context) (bool, error) {
 			err2 := client.CreateDNSRecord(ctx,
+				string(in.InstallConfig.Config.Publish),
```

Comment:
Why cast to string and then cast the constants to string as well in the `CreateDNSRecord` function?
```suggestion
				in.InstallConfig.Config.Publish,
```